modules/client-common/src/main/java/org/apache/ignite/client/proto/query/event/JdbcQueryFetchResult.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.proto.query.event;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.ignite.client.proto.ClientMessagePacker;
+import org.apache.ignite.client.proto.ClientMessageUnpacker;
+import org.apache.ignite.internal.tostring.S;
+
+/**
+ * JDBC query fetch result.
+ */
+public class JdbcQueryFetchResult extends JdbcResponse {
+    /** Query result rows. */
+    private List<List<Object>> items;
+
+    /** Flag indicating the query has no unfetched results. */
+    private boolean last;
+
+    /**
+     * Default constructor is used for deserialization.
```

Comment:
Do we serialize those classes? Where does it happen?

Reply:
Every request and result is serialized in readBinary\writeBinary methods. So that is why I need this constructors.

Reply:
Every request and result is created either from a set of values, or from Unpacker. We could make all fields `final` and introduce two constructors (one with all args, another with Unpacker).

I don't insist, just a suggestion. Immutable classes are easier to reason about.